### PR TITLE
Login page

### PR DESCRIPTION
**Pull request by vfurmane**

### CODE REVIEW

File: .env.template [hunk 1]
```diff
@@ -4,10 +4,13 @@ DATABASE_USERNAME=
 DATABASE_PASSWORD=
 DATABASE_PORT=5432
 
-# api
+# globals
 FRONTEND_BASE_URL=http://localhost:8080
+BACKEND_BASE_URL=http://localhost:3000
+
+FT_OAUTH2_CLIENT_ID=
```
Comment (deblokc) [inline .env.template:11]: i guess you wanted to write 'SECRET' instead of 'ID'

File: README.md [hunk 2]
```diff
@@ -42,11 +42,14 @@ DATABASE_USERNAME=
 DATABASE_PASSWORD=
 DATABASE_PORT=5432
 
-# api
+# globals
 FRONTEND_BASE_URL=http://localhost:8080
+BACKEND_BASE_URL=http://localhost:3000
+
+FT_OAUTH2_CLIENT_ID=
```
Comment (deblokc) [inline README.md:49]: same here

**Review by deblokc — APPROVED** [review]

File: .env.template (hunk 1 above)
Comment (soleraeus) [reply .env.template:11]: Agreed, one of them should probably be SECRET

File: apps/web/pages/index.tsx [hunk 3]
```diff
@@ -1,6 +1,23 @@
...
+import { useEffect, useState } from "react";
 import Home from "./home";
+import { Loading } from "../components/Loading";
 
 export default function Web(): JSX.Element {
+  const router = useRouter();
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const access_token = localStorage.getItem("access_token");
```
Comment (soleraeus) [inline apps/web/pages/index.tsx:11]: No, no and no. Never put a jwt in Local Storage. Ever. Just a sample reading, put I have many more, including OWASP if you are really motivated : https://www.rdegges.com/2018/please-stop-using-local-storage/

File: apps/web/components/LoginForm.tsx [hunk 4]
```diff
@@ -1,24 +1,39 @@
...
+import Link from "next/link";
 
 interface LoginFormData {
   username: string;
   password: string;
 }
 
-async function login(data: LoginFormData): Promise<string | null> {
+function obtainState(): string {
+  let state = localStorage.getItem("state");
```
Comment (soleraeus) [inline apps/web/components/LoginForm.tsx:18]: If it is sensitive information, remove it from local storage

File: apps/web/components/TfaForm.tsx [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    setFormError("");
+    setFormSuccess("");
+
+    await checkTfa(data, props.state)
+      .then((accessToken) => {
+        if (accessToken === null) {
+          throw new Error("An unexpected error occured...");
+        } else {
+          setFormSuccess("Success! Redirecting...");
+          localStorage.setItem("access_token", accessToken);
```
Comment (soleraeus) [inline apps/web/components/TfaForm.tsx:61]: No jwt in local storage

**Review by soleraeus — CHANGES_REQUESTED** [review]
Sensitive items should never be in local storage. I don't understand why you haven't used tools such as next-auth which are in the next's documentation and handle much of the process for you through synchronized sessions and cookies

**Review by savosaicic — APPROVED** [review]

**Review by maxenceeudier — APPROVED** [review]

**Comment by vfurmane** [comment]
Did not fix localStorage issue.

File: apps/web/components/LoginForm.tsx (hunk 4 above)
Comment (vfurmane) [reply apps/web/components/LoginForm.tsx:18]: It is not, I think this can go inside the local storage.

**Pull request merged**